### PR TITLE
ak09916: fixed driver not reporting magnetometer as external

### PR DESCRIPTION
**Pull request by RomanBapst**
Tested on Pixhawk cube + Here 2 gps.
Magnetometer is now detected as external and QGC gives the option to specify the mag orientation.

image

### CODE REVIEW

**Comment by RomanBapst** [comment]
Fixes #12548

**Comment by RomanBapst** [comment]
@davids5 FYI

**Comment by julianoes** [comment]
@RomanBapst so this driver is only ever used for the external Here 2 GPS and never internal?

File: src/drivers/magnetometer/ak09916/ak09916.cpp [hunk 1]
```diff
@@ -252,6 +252,7 @@ AK09916::measure()
 		}
 
 		_px4_mag.set_error_count(perf_event_count(_mag_errors));
+		_px4_mag.set_external(external());
 		_px4_mag.update(timestamp_sample, raw_data.x, raw_data.y, raw_data.z);
```
Comment (davids5) [inline src/drivers/magnetometer/ak09916/ak09916.cpp]: @RomanBapst - That all make sense now. Good catch!

**Comment by VinniF** [comment]
Hi, I have the same issue that the Here2 is not detected as a magnetometer
our (older) firmware version does not seem to include this driver yet. Can I just pull the folder from the repo and paste it into my firmware and add the folder to the cmake? 
grafik

grafik

**Pull request merged**